spec/std/string_spec.cr

```diff
@@ -541,9 +541,22 @@ describe "String" do
     it { "ﬀ".downcase(Unicode::CaseOptions::Fold).should eq("ff") }
     it { "tschüß".downcase(Unicode::CaseOptions::Fold).should eq("tschüss") }
     it { "ΣίσυφοςﬁÆ".downcase(Unicode::CaseOptions::Fold).should eq("σίσυφοσfiæ") }
+
+    describe "with IO" do
+      it { with_io_memory { |io| "HELLO!".downcase io }.should eq "hello!" }
```

Comment:
I introduced `with_io_memory` in a recent commit and was suggested to use `String.build`. Can we use that instead? Totally optional though, I can send a separate PR to fix that later (there's no need for that helper function).

Reply:
Oo good call.

Reply:
Thank you! 🙏